storage/innobase/fil/fil0crypt.cc

```diff
@@ -837,8 +879,12 @@ byte *fil_parse_write_crypt_data_v1(space_id_t space_id, byte *ptr,
   fil_encryption_t encryption = (fil_encryption_t)mach_read_from_1(ptr);
   ptr += 1;
 
-  fil_space_crypt_t *crypt_data = fil_space_create_crypt_data(
-      encryption, key_id, Crypt_key_operation::FETCH_OR_GENERATE_KEY);
+  // since we are parsing v1 crypt_data - it means that encryption of this table
+  // has not yet started and thus we might need to create a key with uuid in
+  // keyring for this table now
+  fil_space_crypt_t *crypt_data =
+      fil_space_create_crypt_data(encryption, key_id, server_uuid,
+                                  Crypt_key_operation::FETCH_OR_GENERATE_KEY);
```

Comment:
Well, true. My thinking was that we always try to be in a state that all necessary keys are already in keyring when we decide to encrypt a tablespace. I guess it seems safer to me than creating key just before encryption starts. Anyways, this is more of a "safer" path and I can remove it if you do not like this approach ?

Reply:
will this be relevant if we stop supporting ENCRYPTION_KEY_ID tables from previous versions?

Also, if this 'safety' path indicates something wrong happenned (like keyring changed or key missing?) it is not got good to silently recreate keys.

Or may be warn user and then recreate?

Reply:
Maybe I understood your first question incorrectly. I will try to answer the first question once again. We have a table that has crypt_data v1 assigned. This table is unencrypted - otherwise the upgrade would have failed. We have turned on encryption threads. The encryption key that is needed to encrypt this table will not be in keyring. We need to create this key now.

Reply:
I can't see how.. Can you list the steps.

1. Execute these SQLs (8.0.18 or below)
2. upgrade to mysqld (which has this patch)

Can you list the SQLs?

Reply:
As per slack discussion, understood that  Keys without UUID are not created/read/written.

Request to add assert that all keys are V2 format and with valid UUID.

Reply:
I have added assert to fill_key_name functions. All KEYRING InnoDB keys need to go through this function. I check there that uuid is never empty. This means that all keys created/read/written in the server need to have uuid set.